kratos/python/add_factories_to_python.cpp

```diff
@@ -21,29 +21,34 @@
 #include "python/add_factories_to_python.h"
 #include "factories/linear_solver_factory.h"
 #include "factories/preconditioner_factory.h"
+#include "factories/register_factories.h"
 
 namespace Kratos
 {
 
 namespace Python
 {
 
+typedef UblasSpace<double, CompressedMatrix, boost::numeric::ublas::vector<double>> SpaceType;
+typedef UblasSpace<double, Matrix, Vector> LocalSpaceType;
+typedef LinearSolver<SpaceType,  LocalSpaceType> LinearSolverType;
+typedef TUblasSparseSpace<std::complex<double>> ComplexSpaceType;
+typedef TUblasDenseSpace<std::complex<double>> ComplexLocalSpaceType;
+
+typedef LinearSolverFactory< SpaceType, LocalSpaceType > LinearSolverFactoryType;
+typedef LinearSolverFactory< ComplexSpaceType, ComplexLocalSpaceType > ComplexLinearSolverFactoryType;
+typedef PreconditionerFactory< SpaceType, LocalSpaceType > PreconditionerFactoryType;
+typedef ExplicitBuilder< SpaceType, LocalSpaceType > ExplicitBuilderType;
+typedef Factory< ExplicitBuilderType > ExplicitBuilderFactoryType;
+typedef ExplicitSolvingStrategy< SpaceType, LocalSpaceType > ExplicitSolvingStrategyType;
+typedef Factory< ExplicitSolvingStrategyType, ModelPart > ExplicitStrategyFactoryType;
```

Comment:
```suggestion
typedef Factory< ExplicitSolvingStrategyType > ExplicitStrategyFactoryType;
```

Reply:
That would not work, I need the auxiliar class, otherwise the variadic template will be not able to deduce types